lib/blocks.php

```diff
@@ -193,6 +196,105 @@ function gutenberg_register_core_block_styles( $block_name ) {
 	}
 }
 
+/**
+ * Change the way styles get loaded depending on their size.
+ *
+ * Optimizes performance and sustainability of styles by inlining smaller stylesheets.
+ *
+ * @return void
+ */
+function gutenberg_maybe_inline_styles() {
+
+	$total_inline_limit = 20000;
+	/**
+	 * The maximum size of inlined styles in bytes.
+	 *
+	 * @param int $total_inline_limit The file-size threshold, in bytes.
+	 * @return int
+	 */
+	$total_inline_limit = apply_filters( 'styles_inline_size_limit', $total_inline_limit );
+
+	global $wp_styles;
+	$styles = array();
+
+	// Build an array of styles that have a path defined.
+	foreach ( $wp_styles->queue as $handle ) {
+		if ( wp_styles()->get_data( $handle, 'path' ) && file_exists( $wp_styles->registered[ $handle ]->extra['path'] ) ) {
+			$block_styles = false;
+			$styles_size  = filesize( $wp_styles->registered[ $handle ]->extra['path'] );
+
+			// Minify styles and get their minified size if SCRIPT_DEBUG is not enabled.
+			if ( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) {
+				// Get the styles and minify them by removing comments & whitespace.
+				$block_styles = gutenberg_get_minified_styles( file_get_contents( $wp_styles->registered[ $handle ]->extra['path'] ) );
+				// Get the styles size.
+				$styles_size = strlen( $block_styles );
+			}
+
+			$styles[] = array(
+				'handle' => $handle,
+				'path'   => $wp_styles->registered[ $handle ]->extra['path'],
+				'size'   => $styles_size,
+				'css'    => $block_styles,
+			);
+		}
+	}
+
+	if ( ! empty( $styles ) ) {
+		// Reorder styles array based on size.
+		usort(
+			$styles,
+			function( $a, $b ) {
+				return ( $a['size'] <= $b['size'] ) ? -1 : 1;
+			}
+		);
+
+		/**
+		 * The total inlined size.
+		 *
+		 * On each iteration of the loop, if a style gets added inline the value of this var increases
+		 * to reflect the total size of inlined styles.
+		 */
+		$total_inline_size = 0;
+
+		// Loop styles.
+		foreach ( $styles as $style ) {
+
+			// Size check. Since styles are ordered by size, we can break the loop.
+			if ( $total_inline_size + $style['size'] > $total_inline_limit ) {
+				break;
+			}
+
+			// Get the styles if we don't already have them.
+			$style['css'] = $style['css'] ? $style['css'] : file_get_contents( $style['path'] );
```

Comment:
When `SCRIPT_DEBUG` is set to `true` then it would read the file. Why it happens here? To minimize the number of read operations?

Reply:
Exactly :smiley_cat: I wanted to avoid the extra read operations that would have occurred if it was performed in the 1st loop. If `SCRIPT_DEBUG` is `false` then we need to read all files and minify them in order to get their _minified_ style. With `SCRIPT_DEBUG` set to `true` we can break out of the loop earlier so it made sense to do it like this.